src/test/kotlin/cz/petrbalat/spring/mvc/test/dsl/controller/DslControllerTest.kt

```diff
@@ -110,16 +128,16 @@ class DslControllerTest : MockMvcProvider {
 
             expect {
                 status { isBadRequest }
+                json("""{"surname":"Jack"}""")
+                json("""{"surname":"Jack", "extraName":"Extra Things"}""", strict = true)
             }
             expect { "$.surname" jsonPathIs "Jack" } //builder,actions, and expects can be called multiple times
         }
     }
 
     @Test
-    fun `minimal call, builder, and expectation`() {
-        mockMvc.perform(HttpMethod.GET, "/hello") {
-            builder { param("name", "world") }
-            expect { status { isOk } }
-        }
+    fun `minimal call, builder, and expectation`() = mockMvc.performGet( "/hello") {
```

Comment:
How useful is the `provider` interface? Should we remove it? Notice how this line is just as compact as the provider equivalent, with the addition of the mockMvc variable. (I doubt even this is very useful since I typically add some mocking calls before and after the mockMvc calls).